perf/hyperloglog.cc

```diff
@@ -0,0 +1,165 @@
+#include "salad/hll.h"
+
+#include <benchmark/benchmark.h>
+#include <cstddef>
+#include <cstdlib>
+#include <cstdint>
+#include <functional>
+#include <vector>
+#include <algorithm>
+
+static constexpr int64_t
+n_registers(int prec)
+{
+	return INT64_C(1) << prec;
+}
+
+uint64_t
+big_rand()
+{
+	uint64_t r1 = std::rand();
+	uint64_t r2 = std::rand();
+	uint64_t r3 = std::rand();
+	return r1 * r2 * r3;
```

Comment:
Actually that's not fair random. For example the lowest bit will be 0 in 87.5% cases.

Reply:
This function is used for expansion of possible values returned by `rand()`. Hashing helps to randomize these values.

Reply:
I applied Standard Mersenne twister from STL.